File: php/abc153_d.php
<?php

##### すげー
function getCount($h)
{
    $count = 1;

    if ($h > 1) {
        # モンスターの体力が1より大きかったら、割る２の
        $count += getCount(floor($h / 2)) * 2;
    }
    return $count;
}

// fscanf(STDIN, "%d", $h);
// echo getCount($h);


###
// fscanf(STDIN, "%d", $h);

// $cnt = 0;
// $ans = 1;
// while ($h > 1) {
//     $h = floor($h /2);
//     $cnt++;
//     $ans = $ans + 2 ** $cnt;
// }
// echo $ans;


#######
fscanf(STDIN, "%d", $n);
$hp = $n;
$num = 1;
$ans = 0;
while($hp > 0){

    // intdiv — 整数値の除算
    $hp = intdiv($hp, 2);
    $ans += $num;
    $num *= 2;

}

echo $ans;

File: ruby/abc153_d.rb

def get_count(h)
    count = 1

    if (h > 1)
        # モンスターの体力が1より大きかったら、割る２の
        count += get_count((h / 2).floor) * 2
    end
    count
end

h = gets.to_i
puts get_count(h);



#############
# h = gets.to_i
#
# ans = 0
# add = 1
#
# while h > 0
#   ans += add
#   add *= 2
#   h = h / 2
# end
# p ans
